splunk-otel-android/src/main/java/com/splunk/rum/SimpleNetworkDetector.java

```diff
@@ -31,20 +31,26 @@
 
     @Override
     public CurrentNetwork detectCurrentNetwork() {
-        NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo(); // Deprecated in API 29
-        if (activeNetwork == null) {
-            return NO_NETWORK;
-        }
-        switch (activeNetwork.getType()) {
-            case ConnectivityManager.TYPE_MOBILE:  // Deprecated in API 28
-                return new CurrentNetwork(NetworkState.TRANSPORT_CELLULAR, activeNetwork.getSubtypeName());
-            case ConnectivityManager.TYPE_WIFI:  // Deprecated in API 28
-                return new CurrentNetwork(NetworkState.TRANSPORT_WIFI, activeNetwork.getSubtypeName());
-            case ConnectivityManager.TYPE_VPN:
-                return new CurrentNetwork(NetworkState.TRANSPORT_VPN, activeNetwork.getSubtypeName());
+        try {
+            NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo(); // Deprecated in API 29
+            if (activeNetwork == null) {
+                return NO_NETWORK;
+            }
+            switch (activeNetwork.getType()) {
+                case ConnectivityManager.TYPE_MOBILE:  // Deprecated in API 28
+                    return new CurrentNetwork(NetworkState.TRANSPORT_CELLULAR, activeNetwork.getSubtypeName());
+                case ConnectivityManager.TYPE_WIFI:  // Deprecated in API 28
+                    return new CurrentNetwork(NetworkState.TRANSPORT_WIFI, activeNetwork.getSubtypeName());
+                case ConnectivityManager.TYPE_VPN:
```

Comment:
Outside the scope of this specific change, but static imports on these would be nice for readability.

Reply:
I think when I wrote this originally, I wanted to make sure it was clear which of these things were Android's, and which were ours, and keeping the class name prefixes made that a little more clear to me.